crawler/main.py

```diff
@@ -62,89 +64,88 @@ def run(sftp: bool, settings_module: str = "") -> None:
-                unique=True,
-            )
-
-            errors: List[str] = []
-            docs_inserted: int = 0
-            latest_file_name: str = ""
-            for centre in centres:
-                logger.info("*" * 80)
-                logger.info(f"Processing {centre['name']}")
-
-                errors.clear()
-                docs_inserted = 0
-                latest_file_name = ""
-                try:
-                    if sftp:
-                        download_csv_files(config, centre)
-
-                    if "merge_required" in centre.keys() and centre["merge_required"]:
-                        master_file_name = merge_daily_files(config, centre)
-
-                        # only upload to SFTP if config explicitly says so - this is to prevent
-                        #   accidental uploads from non-production envs
-                        if config.SFTP_UPLOAD:  # type: ignore
-                            upload_file_to_sftp(config, centre, master_file_name)
-
-                    latest_file_name, errors, docs_to_insert = parse_csv(config, centre)
-
-                    logger.debug(f"Attempting to insert {len(docs_to_insert)} docs")
-                    result = samples_collection.insert_many(docs_to_insert, ordered=False)
-
-                    docs_inserted = len(result.inserted_ids)
-                except BulkWriteError as e:
-                    # This is happening when there are duplicates in the data and the index prevents
-                    #   the records from being written
-                    logger.warning(
-                        f"{e} - usually happens when duplicates are trying to be inserted"
-                    )
-                    docs_inserted = e.details["nInserted"]
-                    write_errors = {write_error["code"] for write_error in e.details["writeErrors"]}
-                    for error in write_errors:
-                        num_errors = len(
-                            list(filter(lambda x: x["code"] == error, e.details["writeErrors"]))
+            with safe_collection(db, COLLECTION_SAMPLES, timestamp) as samples_collection:
+                logger.debug(
+                    f"Creating index '{FIELD_PLATE_BARCODE}' on '{samples_collection.full_name}'"
+                )
+                samples_collection.create_index(FIELD_PLATE_BARCODE)
+                logger.debug(f"Creating compund index on '{samples_collection.full_name}'")
+                # create compound index on 'Root Sample ID', 'RNA ID', 'Result', 'Lab ID' - some data
+                #   had the same plate tested at another time so ignore the data if it is exactly the
+                #   same
+                samples_collection.create_index(
+                    [
+                        (FIELD_ROOT_SAMPLE_ID, pymongo.ASCENDING),
+                        (FIELD_RNA_ID, pymongo.ASCENDING),
+                        (FIELD_RESULT, pymongo.ASCENDING),
+                        (FIELD_RNA_ID, pymongo.ASCENDING),
```

Comment:
`FIELD_RNA_ID` is duplicate with line 80

Reply:
Thanks, order of indicies is important, but it looks like the first definition is taking precedence here (based on observing the actual index that gets created) so I should be able to drop the second index.